src/Tokens/TokenRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Statamic\Eloquent\Tokens;
+
+use Statamic\Contracts\Tokens\Token as TokenContract;
+use Statamic\Tokens\TokenRepository as Repository;
+
+class TokenRepository extends Repository
+{
+    public function find(string $token): ?Token
+    {
+        $model = app('statamic.eloquent.tokens.model')::whereToken($token)->first();
+
+        if (! $model) {
+            return null;
+        }
+
+        return Token::fromModel($model);
+    }
+
+    public function save(TokenContract $token): bool
```

Comment:
I did run into an issue here (and in `delete`). This hint should really be `Statamic\Eloquent\Tokens\Token`, since it uses `toModel`, but if I set it to that it complained it wasn't compatible with the base class. Not sure if I was doing it wrong or PHP was just being weird.

Reply:
`Statamic\Eloquent\Tokens\Token` does implement `TokenContract` so that's fine. You can call `toModel` on it. I think maybe just your IDE would be complaining that the `toModel` method didn't exist since it's not in the contract. Should be fine though.

You can probably add a docblock to get that IDE support.

```php
/**
 * @param \Statamic\Eloquent\Tokens\Token $token
 */
public function save(TokenContract $token): bool
```